index.js

```diff
@@ -73,7 +73,11 @@ const _splitToStream = (outStreamCreate, fileStream, partitionStreamSize, callba
 	});
 };
 
-const split = (fileStream, maxFileSize, rootFilePath, callback) => {
+const split = (fileStream, maxFileSize, rootFilePath, callback) => _split(fileStream, maxFileSize, rootFilePath, generateFilePath, callback);
```

Comment:
It seems a bit ambiguous that the `generateFilePath` being passed to the `_split` call refers to a default function. What do you think about naming it to something like `defaultGenerateFilePath`? I'm also starting to notice that a name like `getFilePath` is probably more suitable than `generateFilePath`.

Reply:
The problem I see is that we have defined a global variable and then defined a local variable in `getSplitWithGenFilePath` with the same name. The definitive outcome is that JavaScript will use the local variable when in the local scope of the function, but I can see this being a point of potential confusion in terms of readability and also best practices.

Also, another way we can fix this is similar to my suggestion above. We can quite literally move line 5 right above the call to _split, so that `generateFilePath` is a local variable. Any of the 3 solutions I listed will fix this problem, and I'm sure there are more (even possibly better) solutions!

Reply:
btw the tests break if the output dir in test isnt created before running `npm t`

Reply:
Ahh, okay I'll fix the test breakage later. Thanks for pointing that out.